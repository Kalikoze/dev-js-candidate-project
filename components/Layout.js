import React from 'react';
import PropTypes from 'prop-types';

const Layout = ({ children }) => (
  <section className='layout'>
    <video src='../static/ambient-background.mp4' autoPlay loop />
    <audio src='../static/ambiance.mp3' autoPlay loop/>
    <section className='container'>
      {children}
    </section>
    <style jsx>{`
      .layout {
        display: flex;
        height: 100vh;
        width: 100%;
      }

      video {
        height: 100%;
        object-fit: fill;
        width: 100%;
      }

      .container {
        display: flex;
        flex-wrap: wrap;
        height: 100vh;
        justify-content: center;
        position: absolute;
        width: 100%;
      }
    `}</style>
  </section>
);

Layout.propTypes = {
  children: PropTypes.array.isRequired,
};

export default Layout;
