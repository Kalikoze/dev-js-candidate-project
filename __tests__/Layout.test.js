import React from 'react';
import Layout from '../components/Layout';
import renderer from 'react-test-renderer';
// import 'jest-styled-components';

describe('Layout component', () => {
  it('should render the Layout correctly', () => {
    const tree = renderer.create(<Layout>
      <video src='../static/ambient-background.mp4' autoPlay loop />
      <section className='container'>
      </section>
    </Layout>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
