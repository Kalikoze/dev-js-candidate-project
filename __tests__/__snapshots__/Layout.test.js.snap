// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component should render the Layout correctly 1`] = `
<section
  className="jsx-2146746531 layout"
>
  <video
    autoPlay={true}
    className="jsx-2146746531"
    loop={true}
    src="../static/ambient-background.mp4"
  />
  <section
    className="jsx-2146746531 container"
  >
    <video
      autoPlay={true}
      loop={true}
      src="../static/ambient-background.mp4"
    />
    <section
      className="container"
    />
  </section>
</section>
`;
